### PR TITLE
Rss ecomm 3 12 && rss ecomm 3 13

### PR DESCRIPTION
**Pull request by JoMarianek**
1. Task 3_12: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_12.md)
2. Task: 3_13: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_13.md)
3. Screenshot: 
image
4. Change description: Implement modal carousel, where users can cycle through enlarged version of images.
The enlarged image is a bit cut off, see screenshot but I couldn't get the formatting right for the life of me.

### CODE REVIEW

File: src/views/DetailedPageView.vue [hunk 1]
```diff
@@ -27,6 +29,12 @@
                         (elem) => elem.value.currencyCode === currencyCode
                     );
                     return priceObj ? (priceObj.value.centAmount / 100).toFixed(2) : 'N/A';
+                },
+                showOverlay(index: number) {
+                    console.log('Image clicked:', index);
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:34]: Please remove console outputs.

File: src/views/DetailedPageView.vue [hunk 2]
```diff
@@ -27,6 +29,12 @@
...
+                    this.currentImageIndex = index;
+                    this.overlayVisible = true;
+                    console.log('Overlay visibility:', this.overlayVisible);
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:37]: Please remove console outputs.

File: src/views/DetailedPageView.vue [hunk 3]
```diff
@@ -27,6 +29,12 @@
                         (elem) => elem.value.currencyCode === currencyCode
                     );
                     return priceObj ? (priceObj.value.centAmount / 100).toFixed(2) : 'N/A';
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:31]: Please add line break before the return, it helps readability a lot.

Also replace single quotes ' with doubles quotes " for consistency with the rest of the project.

File: src/views/DetailedPageView.vue [hunk 4]
```diff
@@ -45,6 +53,14 @@
                         };
                     }) || [];
                 },
+                extraLargeImages() {
+                    return this.product.masterData?.current?.masterVariant?.images.map(image => {
+                        return {
+                            ...image,
+                            url: image.url.replace(/(\.[\w\d_-]+)$/i, '-zoom$1')
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:60]: Also replace single quotes ' with doubles quotes " for consistency with the rest of the project.

File: src/views/DetailedPageView.vue [hunk 5]
```diff
@@ -85,6 +102,24 @@
...
+                <v-btn @click="overlayVisible = false">
+                    x
+                </v-btn>
+                <v-carousel :show-arrows="showArrows">
+                    <v-carousel-item 
+                            v-for="(image, index) in largeImages"
+                            :key="index"
+                            class="modal-carousel-item"
+                            :src="image.url"
+                        >
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:117]: Indents here seem a bit off.

File: src/views/DetailedPageView.vue [hunk 6]
```diff
@@ -85,6 +102,24 @@
...
+                            <v-img :src="image.url" class="overlay-image" alt="Enlarged Product Image" aspect-ratio="1"></v-img>
+                        </v-carousel-item>
+                    </v-carousel>
```
Comment (vetalapo) [inline src/views/DetailedPageView.vue:120]: Indents here seem a bit off.

**Review by vetalapo — CHANGES_REQUESTED** [review]

File: src/views/DetailedPageView.vue (hunk 5 above)
Comment (JoMarianek) [reply src/views/DetailedPageView.vue:116]: Its the props for the v-carousel-item. i think they are supposed to be indented when listed individually on a line. What change would you suggest?

Comment (JoMarianek) [reply src/views/DetailedPageView.vue:116]: nevermind found the issue

File: src/views/DetailedPageView.vue [hunk 7]
```diff
@@ -85,6 +101,24 @@
...
+                        :key="index"
+                        class="modal-carousel-item"
+                        :src="image.url"
+                    >
+                        <v-img :src="image.url" class="overlay-image" alt="Enlarged Product Image" aspect-ratio="1"></v-img>
+                    </v-carousel-item>
+                </v-carousel>
+            </v-row>
+        </v-overlay>
     </v-container>
```
Comment (valeryiay) [inline src/views/DetailedPageView.vue:122]: add
.wrapper-container {
    overflow-x: hidden;
    max-width: 1600px;
}
and content will not go beyond the viewport
Screenshot 2024-06-04 231719

**Review by valeryiay — CHANGES_REQUESTED** [review]
enlarged image modal works, good job!

File: src/views/DetailedPageView.vue (hunk 7 above)
Comment (JoMarianek) [reply src/views/DetailedPageView.vue:122]: Oh, thanks for the tip!

**Review by valeryiay — APPROVED** [review]
Super! thank you😊

**Review by vetalapo — APPROVED** [review]

**Pull request merged**